backend/src/apis/round.rs

```diff
@@ -44,6 +44,11 @@ impl SolvencyProof {
     }
 }
 
+/*
+   Since we are now using the `MstInclusionProof` only for on-chain verification,
+   why don't we replace the data type with something that can be directly consumed by the contract?
+   Similar to `SolvencyProof`.
```

Comment:
I think we can set the on-chain verifcation (via the smart contract interface) as standard way to perform the proof of inclusion verification. This is because of the nice UX that it provides + full transparency on the code being executed compared to the other available solutions.

Therefore I suggest to remove every dependency that relates to off-chain verification and foucs on the on-chain path.